### PR TITLE
RAD-272 Support for radiology report templates

### PR DESCRIPTION
**Pull request by ivange94**
RAD-272 Support for radiology report templates

## Description
implemented java pojo to represent templates and also created DAOs and well tested Service methods to save, get and delete templates. Also started with implementation of the templates tab, you can see it a screenshot here, http://picpaste.com/for_github-P2t4gp8a.png

see https://issues.openmrs.org/browse/RAD-272

## Checklist:

- [x] My pull request only contains one single commit.
- [x] My pull request is based on the latest master branch
  `git pull --rebase upstream master`.
- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/radiology/report/template/web/RadiologyDashboardReportTemplatesTabPortletController.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+@Controller
+@RequestMapping("**/reportTemplatesTab.portlet")
+public class RadiologyDashboardReportTemplatesTabPortletController extends PortletController {
+    
+    // declaration of service should go here
+    
+    @Override
+    protected void populateModel(HttpServletRequest request, Map<String, Object> model) {
+        model.put("username", "ivange94");
```
Comment (teleivo) [inline omod/src/main/java/org/openmrs/module/radiology/report/template/web/RadiologyDashboardReportTemplatesTabPortletController.java]: What is this for?

Comment (ivange94) [inline omod/src/main/java/org/openmrs/module/radiology/report/template/web/RadiologyDashboardReportTemplatesTabPortletController.java]: was just testing that the controller communicates with the view fine. Forgot to remove it in my final commit, will do so now.

File: api/src/main/resources/liquibase.xml [hunk 2]
```diff
@@ -338,4 +338,47 @@
...
+			<column name="dcterms_date" type="varchar(32)"/>
+			<column name="dcterms_creator" type="varchar(128)"/> 
+		</createTable>
+	</changeSet>
+	<changeSet id="radiology-31" author="ivange94">
+		<comment>Add privilege to work with radiology report templates</comment>
+		<insert tableName="privilege">
+			<column name="privilege" value="Add Radiology Report Templates"/>
+			<column name="description" value="Able to add new report templates"/>
+			<column name="uuid" value="ae4eba5c5604d05828e2463707215802"></column>
```
Comment (bgeVam) [inline api/src/main/resources/liquibase.xml]: is there an intention behind this uuid format? if not, please use the format containing the "-" character, you may take a look at the other changeSets for examples

Comment (ivange94) [inline api/src/main/resources/liquibase.xml]: @bgeVam i looked at your own code before implementing mine but i did not know how you generated the uuid, so i just used the hashcode of the priviliges. Is there a pattern to the way you generted the uuids?

Comment (teleivo) [inline api/src/main/resources/liquibase.xml]: @ivange94 use `java.util.UUID.randomUUID()`

File: api/src/main/java/org/openmrs/module/radiology/report/template/HibernateMrrtReportTemplateDAO.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    
+    /**
+     * @see org.openmrs.module.radiology.report.template.MrrtReportTemplateService#getMrrtReportTemplate(Integer)
+     */
+    @Override
+    public MrrtReportTemplate getMrrtReportTemplate(Integer templateId) {
+        return (MrrtReportTemplate) sessionFactory.getCurrentSession()
+                .get(MrrtReportTemplate.class, templateId);
+    }
+
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/module/radiology/report/template/HibernateMrrtReportTemplateDAO.java]: @ivange94 you will need a getMrrtReportTemplateByUuid(String) for the REST service

File: api/src/main/java/org/openmrs/module/radiology/report/template/MrrtReportTemplateException.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.radiology.report.template;
+
+import org.openmrs.api.APIException;
+
+public class MrrtReportTemplateException extends APIException {
```
Comment (teleivo) [inline api/src/main/java/org/openmrs/module/radiology/report/template/MrrtReportTemplateException.java]: @ivange94 I think for now you can delete your custom MrrtReportTemplateException. it just adds more code. if later we will see that its truly needed, lets create it

File: api/src/main/java/org/openmrs/module/radiology/report/template/HibernateMrrtReportTemplateDAO.java (hunk 3 above)
Comment (ivange94) [inline api/src/main/java/org/openmrs/module/radiology/report/template/HibernateMrrtReportTemplateDAO.java]: could find a sample implementation for that here, https://github.com/teleivo/openmrs-module-radiology/blob/RAD-140/api/src/main/java/org/openmrs/module/radiology/order/RadiologyOrderDAO.java

Comment (teleivo) [inline api/src/main/java/org/openmrs/module/radiology/report/template/HibernateMrrtReportTemplateDAO.java]: check openmrs-core they have it in their OrderService

File: api/pom.xml [hunk 5]
```diff
@@ -47,6 +47,12 @@
 			<artifactId>openmrs-test</artifactId>
 			<type>pom</type>
 		</dependency>
+		<dependency>
```
Comment (teleivo) [inline api/pom.xml]: No need for this comment

**Comment by ivange94** [comment]
will create another, accidentally introduced other commits in this pull request.

**Comment by ivange94** [comment]
@teleivo i tried pushing a single commit but accidentally pushed all your uncommitted changes on RAD-140 to this branch. New pull request is, https://github.com/openmrs/openmrs-module-radiology/pull/304. I will resolve all the comments you made on the previous pull request there. :)

**Pull request closed without merging**